lib/appmap/hook/method.rb

```diff
@@ -69,28 +69,24 @@ def activate
         after_hook = self.method(:after_hook)
         with_disabled_hook = self.method(:with_disabled_hook)
 
-        hook_method_def = Proc.new do |*args, &block|
-          is_array_containing_empty_hash = ->(obj) {
-            obj.is_a?(Array) && obj.length == 1 && obj[0].is_a?(Hash) && obj[0].size == 0
-          }
-
-          call_instance_method = -> {
-            # https://github.com/applandinc/appmap-ruby/issues/153
-            if NEW_RUBY && is_array_containing_empty_hash.(args) && hook_method.arity == 1
-              if NEW_RUBY
-                hook_method.bind_call(self, {}, &block)
-              else
-                hook_method.bind(self).call({}, &block)
-              end
+        is_array_containing_empty_hash = ->(obj) {
+          obj.is_a?(Array) && obj.length == 1 && obj[0].is_a?(Hash) && obj[0].size == 0
```

Comment:
I wonder, wouldn't it work to replace this method with `args == [{}]`?

Reply:
Maybe but, I'm afraid to change this.